### PR TITLE
Use name instead of data-bz-retained

### PR DESCRIPTION
**Pull request by rshipp**
### Description

Update everything to use input `name` instead of `data-bz-retained`.

### Motivation and Context

Task: https://app.asana.com/0/1174274412967132/1199178251531117/f

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by sadleb — APPROVED** [review]
I’ll have to defer to Jess on the CKEditor stuff but nothing problematic jumps out to me.

File: app/javascript/ckeditor/contentcommonediting.js [hunk 1]
```diff
@@ -47,13 +46,13 @@ export default class ContentCommonEditing extends Plugin {
...
+            allowAttributes: [ 'type', 'placeholder', 'name' ],
             allowIn: [ '$root', 'industrySelectorContainer' ],
         } );
 
         schema.register( 'textArea', {
             isObject: true,
-            allowAttributes: [ 'placeholder' ],
+            allowAttributes: [ 'placeholder', 'name' ],
             allowIn: [ '$root' ],
         } );
```
Comment (jessouyangbraven) [inline app/javascript/ckeditor/contentcommonediting.js:57]: GitHub isn't letting me comment directly on the lines, but 59-63 for `fileUpload` -- does that need a `name` as well? I think it's an `<input type="file">` under the hood.

File: app/javascript/ckeditor/insertradioquestioncommand.js [hunk 2]
```diff
@@ -18,11 +18,12 @@ export default class InsertRadioQuestionCommand extends Command {
 }
 
 function createRadioQuestion( writer ) {
-    const radioGroup = uid();
+    const uniqueId = new UniqueId();
+    const radioGroup = uniqueId.getNewName();
     // Value must be unique within each group, but otherwise doesn't matter.
     // No reason to tie in retained data or anything, we can just use "1".
```
Comment (jessouyangbraven) [inline app/javascript/ckeditor/insertradioquestioncommand.js:24]: Delete/update comment?

**Comment by sadleb** [comment]
I don’t think we should merge this or the other HTML cleanup PRs until after the demo bc it will likely break Helen’s demo project prototype: https://platform.braven.org/custom_contents/7/edit

**Review by jessouyangbraven — APPROVED** [review]
Just the one question about whether file uploads also need a name. Other than that, this looks good to me!

File: app/javascript/ckeditor/contentcommonediting.js (hunk 1 above)
Comment (jessouyangbraven) [reply app/javascript/ckeditor/contentcommonediting.js:57]: Never mind. You get rid of fileupload in a later PR. Ignore me!

**Pull request merged**